src/components/utils/http.ts

```diff
@@ -28,7 +28,13 @@ export const performHttpReq = <T>(
 
   request.onload = () => {
     if (request.status === 200 || request.status === 201) {
-      onSuccess(JSON.parse(request.response))
+      try {
+        const resp = JSON.parse(request.response)
+        onSuccess(resp)
+      } catch (error) {
+        console.warn('Unparseable response:', request.response)
+        onSuccess({} as T)
```

Comment:
Should this be tracked when it happens? I'm assuming this error is being caught in case back end service does somehow return a non-JSON response instead of what we're expecting.